src/libslic3r/GCode.cpp

```diff
@@ -306,11 +306,12 @@ static std::vector<Vec2d> get_path_of_change_filament(const Print& print)
             _wipe_speed = gcodegen.writer().get_current_speed() / 60.0;
         if(_wipe_speed < 10)
             _wipe_speed = 10;
-
-        // get the retraction length
-        double length = toolchange
-            ? gcodegen.writer().extruder()->retract_length_toolchange()
-            : gcodegen.writer().extruder()->retraction_length();
+        
+        // always attempt wipe with the regular filament retraction amount as tool changes may have retraction lengths that are too large and unecessary for wiping and can also
+        // cause retraction to fail (extruder stall) if wiping over a too small a distance. For a tool change, the remaining retraction distance will be performed in the
+        // parent function.
+        double length = gcodegen.writer().extruder()->retraction_length();
```

Comment:
This will break tool change code.
We should keep the original one

Reply:
How? Because the tool change retraction is performed after, in the parent class anyway?

Reply:
The calling function checks if there is any left over retraction to be made - if there is a tool change included. It then performs it.

Reply:
![IMG_2705](https://github.com/SoftFever/OrcaSlicer/assets/59056762/1768846c-6d67-4ceb-a29e-f29265d8058f)
Here after the retract in wipe command there is the tool change retraction.